common/domain/replication_queue_test.go

```diff
@@ -210,3 +214,111 @@ func TestUpdateAckLevel(t *testing.T) {
 		})
 	}
 }
+
+func TestReplicationQueueImpl_GetAckLevels(t *testing.T) {
+	tests := []struct {
+		name      string
+		want      map[string]int64
+		wantErr   bool
+		setupMock func(q *persistence.MockQueueManager)
+	}{
+		{
+			name:    "successful ack levels retrieval",
+			want:    map[string]int64{"testCluster": 100},
+			wantErr: false,
+			setupMock: func(q *persistence.MockQueueManager) {
+				q.EXPECT().GetAckLevels(gomock.Any()).Return(map[string]int64{"testCluster": 100}, nil)
+			},
+		},
+		{
+			name:    "ack levels retrieval fails",
+			wantErr: true,
+			setupMock: func(q *persistence.MockQueueManager) {
+				q.EXPECT().GetAckLevels(gomock.Any()).Return(nil, errors.New("retrieval error"))
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockQueue := persistence.NewMockQueueManager(ctrl)
+			rq := NewReplicationQueue(mockQueue, "testCluster", nil, nil)
+			tt.setupMock(mockQueue)
+			got, err := rq.GetAckLevels(context.Background())
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.want, got)
+			}
+			ctrl.Finish()
```

Comment:
this finish call is not needed.

Reply:
don't we require ctrl.Finish() to ensures all expected interactions with mock objects were met in a test?